lua/cfc_time/server/storage_options/mysql.lua

```diff
@@ -207,9 +297,37 @@ function storage:GetTotalTime( steamID, callback )
     query:start()
 end
 
-function storage:CreateSession( callback, steamID, sessionStart, sessionEnd, duration )
-    local newSession = self:Prepare( "newSession", callback, steamID, sessionStart, sessionEnd, duration )
-    newSession:start()
+function storage:CreateSession( callback, steamID, sessionStart, sessionEnd, sessionDuration )
+    local maxDuration = self.MAX_SESSION_DURATION
+    local sessionsCount = math.ceil( maxDuration / sessionDuration )
+    if sessionsCount == math.huge then sessionsCount = 1 end
+
+    logger:debug( "[" .. tostring( steamID ) .. "] Creating " .. tostring( sessionsCount ) .. " sessions to accomodate duration of: " .. tostring( sessionDuration ) )
+
+    local transaction = self:InitTransaction()
+    transaction.onSuccess = function()
+        if callback then callback() end
+    end
+
+    local function addSession( duration, newStart, newEnd )
+        local debugLine = "Queueing new session of duration: %d ( start: %d | end: %d )"
+        logger.debug( string.format( debugLine, duration, newStart, newEnd ) )
+
+        local newSession = self:Prepare( "newSession", nil, steamID, newStart, newEnd, duration )
+        transaction:addQuery( newSession )
+    end
+
+    for i = 1, sessionsCount do
+        local usedDuration = maxDuration * ( i - 1 )
+
+        local newDuration = sessionDuration - usedDuration
```

Comment:
```suggestion
        local newDuration = math.min( sessionDuration - usedDuration, maxDuration )
```
Otherwise newDuration is always the total remaining time of the entire session, and thus often greater than maxDuration